pkg/ee/default-application-controller/util.go

```diff
@@ -0,0 +1,39 @@
+/*
+                  Kubermatic Enterprise Read-Only License
+                         Version 1.0 ("KERO-1.0”)
+                     Copyright © 2024 Kubermatic GmbH
+
+   1.	You may only view, read and display for studying purposes the source
+      code of the software licensed under this license, and, to the extent
+      explicitly provided under this license, the binary code.
+   2.	Any use of the software which exceeds the foregoing right, including,
+      without limitation, its execution, compilation, copying, modification
+      and distribution, is expressly prohibited.
+   3.	THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND,
+      EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+      MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+      IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+      CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+      TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+      SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+   END OF TERMS AND CONDITIONS
+*/
+
+package defaultapplicationcontroller
+
+import (
+	"context"
+
+	appskubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/apps.kubermatic/v1"
+)
+
+func GetAppNamespace(ctx context.Context, application appskubermaticv1.ApplicationDefinition) appskubermaticv1.AppNamespaceSpec {
```

Comment:
I think we should use pointers for this. No need to make a copy of the object. TBH I don't know why it was using full structs in the original `generateApplicationInstallation` func, but I think going forward we should conserve memory where we can. This comment also applies to all the other newly added funcs throughout the callstack.
```suggestion
func GetAppNamespace(ctx context.Context, application *appskubermaticv1.ApplicationDefinition) appskubermaticv1.AppNamespaceSpec {
```

Reply:
note: I definitely want this for the func arg.
For the return parameter, I am more flexible. While I would prefer to have it return a pointer, I can live with it returning a copy. One more thing: It it is converted to a pointer in line 33 we need to make a copy of the object first and then return a pointer to the copied object.

Reply:
I changed both, input and return value, to pointers. The address is dereferenced during the generation of an applicationinstallation after the function was called.